### PR TITLE
Proof marshaling

### PR DESCRIPTION
**Pull request by olaure01**
experiment proof marshaling
possible tool for #82

### CODE REVIEW

**Comment by olaure01** [comment]
> However I think it's not a good idea to hash proof like that, since it will very likely break backward compatibility. For example, you save/share many proofs, and we deploy a new feature like Cut. Since Proof object would have changed (with Cut_proof) marshalling may shift and unmarshalling your previously saved proofs may not work.

You are probably right.
My idea was indeed more in the short term:
- I start a huge proof, I stop in the middle, I want to save it and start again tomorrow
- or I build a nice proof, I want to send it to my friend so that he can play with / modify it

File: proof_marshal.ml [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+    Bytes.of_string (Base64.decode_exn s)
+
+let direct_marshal proof =
+    to_base64 (Marshal.to_bytes proof [])
+
+let direct_unmarshal str =
+    Marshal.from_bytes (from_base64 str) 0
+
+let compact_marshal proof =
+    to_base64 (Marshal.to_bytes (get_conclusion proof, to_request_option_list proof) [])
```
[the remark this answers is not in the document]
Comment (etiennecallies) [reply proof_marshal.ml]: indeed 👍🏻

**Comment by etiennecallies** [comment]
> My idea was indeed more in the short term:
> 
> * I start a huge proof, I stop in the middle, I want to save it and start again tomorrow
> * or I build a nice proof, I want to send it to my friend so that he can play with / modify it

I agree with that. It's a share-proof feature not a save-proof feature.

**Comment by olaure01** [comment]
Fixes #82

**Pull request merged**